### PR TITLE
GH-2093 import order formatting

### PR DESCRIPTION
**Pull request by jeenbroekstra**
GitHub issue resolved: #2093 

Briefly describe the changes proposed in this PR:

- added maven plugin to automatically fix order of imports as part of default build cycle.
- modified maven build to automatically apply formatting changes and sorting in local build, instead of just checking and failing.
- CI checks formatting and import sorting quickly before full build, to "fail fast"
- running `mvn impsort:sort` manually will fix import order.
- added instructions to code conventions readme

Note that we will likely need to run this once on the develop branch as well after syncing. 

---- 
PR Author Checklist: 

 - [x] my pull request is [self-contained](https://rdf4j.org/documentation/developer/merge-strategy/#self-contained-changes-pull-requests-and-commits)
 - [ ] I've added tests for the changes I made
 - [x] every commit message starts with the issue number (GH-xxxx) followed by a meaningful description of the change
 - [x] every commit has been [signed off](https://stackoverflow.com/questions/1962094/what-is-the-sign-off-feature-in-git-for)

Note: we merge all feature pull requests using [squash and merge](https://help.github.com/en/github/administering-a-repository/about-merge-methods-on-github#squashing-your-merge-commits). See [RDF4J git merge strategy](https://rdf4j.org/documentation/developer/merge-strategy/) for more details.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -56,6 +56,7 @@ Eclipse RDF4J follows the [Eclipse Coding Conventions for Java](https://wiki.ecl
 
 - We use a line width of 120 characters.
 - We use Unix line endings (LF).
+- We require curly braces for every control statement body (e.g. if-else), even if it is a single line.
```
Comment (jeenbroekstra) [inline .github/CONTRIBUTING.md:59]: Not part of this issue, strictly speaking, but I just realized it might be good to mention this here.

**Comment by hmottestad** [comment]
Could we have it do the sorting during the default install cycle and instead add a verify call as a step in the CI. I find it tedious when I’ve spent 3 minutes waiting for the build just to have it error out because I forgot to format a file.

**Comment by jeenbroekstra** [comment]
Unless I misunderstand, that's exactly what I've done here.

File: .github/workflows/pr-verify.yml [hunk 2]
```diff
@@ -23,11 +23,11 @@ jobs:
...
           ${{ runner.os }}-maven-
     - name: Verify formatting
-      run: mvn -B formatter:validate
+      run: mvn -B formatter:validate impsort:check
     - name: Build
-      run: mvn -B clean install -DskipTests  
+      run: mvn -B clean install -Pquick,\!formatting-check
     - name: Run tests
-      run: mvn -B verify -Pcompliance --file pom.xml
+      run: mvn -B verify -Pcompliance,\!formatting-check --file pom.xml
```
Comment (jeenbroekstra) [inline .github/workflows/pr-verify.yml:30]: I am not 100% the backslash is necessary. You need it if you run this on the command line but I don't know how clever the github actions are with this kind of thing.

**Comment by jeenbroekstra** [comment]
@hmottestad I believe this setup is bit more "fail fast". Unless you have further comments I intend to merge this first thing tomorrow.

**Comment by hmottestad** [comment]
I would prefer if the install phase formatted the files and sorted the imports rather than failed if you forgot to run those two commands first. 

The reason behind the plugins is to enforce code format before someone can merge. Which is already achieved by the GitHub CI.

**Comment by jeenbroekstra** [comment]
Ah I see, you want fix rather than check as part of the local build! Can do.

**Pull request merged**